components/script/dom/request.rs

```diff
@@ -290,15 +291,12 @@ impl Request {
                 return Err(Error::Type("Method is forbidden".to_string()));
             }
             // Step 25.2
-            let method_lower = init_method.to_lower();
-            let method_string = match method_lower.as_str() {
-                Some(s) => s,
+            let method = match init_method.as_str() {
+                Some(s) => normalize_method(s),
```

Comment:
I meant that add `Step 25.3` above this line, and make it `*request.method.borrow_mut() = normalize_method(s)` instead.